### PR TITLE
Auto-confirm init questions and add eviroment variable support

### PR DESCRIPTION
**Pull request by HrisIord**
Changes in this MR:

- Add "-y" flag to the init command. When it's passed it with auto-confirm all the questions about adding directories. This makes the init command more script friendly.
- Add environment variable support for configuration parameters. When there is an environment variables use those to override config file configuration. Also remove error when there is no config file, in this case we will use the defaults along with any environment variable values we find.

### CODE REVIEW

File: src/FlanCommand.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { IConfig } from "@oclif/config";
 import { OutputFlags } from "@oclif/parser";
 import * as path from "path";
 import * as fs from "fs-extra";
+import process from "process";
```
Comment (hhaidar) [inline src/FlanCommand.ts:6]: I don't think this needs to be imported

File: src/FlanCommand.ts [hunk 2]
```diff
@@ -57,39 +58,86 @@ export default abstract class FlanCommand extends Command {
...
+    // get envirment variable configurations
+    let envConfig: {
+      baseDir?: string;
+      saveDir?: string;
+      repoDir?: string;
+      repository?: string;
+    } = {};
+    let envDbConfig: { host?: string; db?: string; user?: string } = {};
+
+    if (process.env.FLAN_DB_HOST) {
```
Comment (hhaidar) [inline src/FlanCommand.ts:70]: it's a bit less code to just use process.env.FLAN_DB_HOST directly like `process.env.FLAN_BASE_DIR || configJSON.baseDir || this.localConfig.baseDir`

File: src/commands/init.ts [hunk 3]
```diff
@@ -10,6 +11,10 @@ export default class Init extends FlanCommand {
 
   static flags = {
     ...FlanCommand.flags,
+    confirm: flags.boolean({
+      char: "y",
```
Comment (hhaidar) [inline src/commands/init.ts:15]: should be `--yes` with `-y` as alias, similar to `apt`: https://askubuntu.com/questions/672892/what-does-y-mean-in-apt-get-y-install-command

File: src/FlanCommand.ts [hunk 4]
```diff
@@ -57,39 +58,86 @@ export default abstract class FlanCommand extends Command {
...
         ...configJSON.database,
+        ...envDbConfig,
       },
     };
 
-    if (configJSON.repoDir) {
-      this.localConfig.repoDir = path.resolve(configJSON.repoDir);
+    if (envConfig.repoDir || configJSON.repoDir) {
+      this.localConfig.repoDir = path.resolve(
+        envConfig.repoDir || configJSON.repoDir || ""
```
Comment (hhaidar) [inline src/FlanCommand.ts:133]: is the empty string intentional?

File: src/FlanCommand.ts [hunk 5]
```diff
@@ -57,39 +58,86 @@ export default abstract class FlanCommand extends Command {
...
+      );
     }
 
-    if (configJSON.saveDir) {
-      this.localConfig.saveDir = path.resolve(configJSON.saveDir);
+    if (envConfig.saveDir || configJSON.saveDir) {
+      this.localConfig.saveDir = path.resolve(
+        envConfig.saveDir || configJSON.saveDir || ""
```
Comment (hhaidar) [inline src/FlanCommand.ts:139]: yeah both of these should have a value according to the conditional, not sure if `|| ""` is needed?

Comment (HrisIord) [reply src/FlanCommand.ts]: So I tried removing it but it seems that TypeScript is not smart enough to figure out that the statement in the path.resolve matches the if statement. When I remove the `|| ""` VS code complains about the parameter possibly being undefined. I agree that the `"" ` should not be needed but I also don't see a cleaner way to get rid of it without tripping up TypeScript.

Comment (hhaidar) [reply src/FlanCommand.ts]: haha that's definitely annoying, if you throw a comment above each one then we should good to go

**Comment by hhaidar** [comment]
Thank you @HrisIord  🎆

**Pull request merged**